test/asn1_internal_test.c

```diff
@@ -70,7 +70,7 @@ static int test_standard_methods()
 
     for (tmp = standard_methods, i = 0; i < OSSL_NELEM(standard_methods);
          i++, tmp++) {
-        if ((*tmp)->pkey_id < last_pkey_id) {
+        if (!TEST_int_ge((*tmp)->pkey_id, last_pkey_id)) {
```

Comment:
It is a failure condition of sorts.  Setting `last_pkey_id` to zero skips the normal exit and prints the method table.  It is up to the user to find the one that is sorted incorrectly.  Adding the message pinpoints this directly.

I'm still happy to revert this change but I think it adds some value.